torch/csrc/jit/register_prim_ops.cpp

```diff
@@ -2341,6 +2341,24 @@ RegisterOperators reg2({
           return 0;
         },
         aliasAnalysisFromSchema()),
+    Operator(
+        "aten::list(Tensor(a) t) -> Tensor[](a)",
```

Comment:
This is close but not quite right. This is saying that the input tensor aliases the output list. The list and input tensor don't alias each other.  The alias annotation we want is:
`aten::list(Tensor(a -> *) t) -> Tensor[]` , which says that the input tensor enters the heap ( the list). We don't track the elements that containers (lists, dictionaries, objects) contain and assume that they all contain the elements in the heap. The list itself is new (doesn't alias any other lists) so it doesn't need an annotation.

Reply:
Done. Although I will probably need to read some source code to understand this.

Reply:
This should help a little bit: https://github.com/pytorch/pytorch/blob/master/torch/csrc/jit/docs/OVERVIEW.md#aliasing-and-mutation-annotations-in-functionschema